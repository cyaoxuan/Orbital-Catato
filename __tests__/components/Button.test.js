import {
    FilterButton,
    PillButton,
    TextButton,
} from "../../app/components/Button";
import { cleanup, fireEvent, render } from "@testing-library/react-native";

afterEach(cleanup);

describe("<PillButton />", () => {
    it("renders successfully", () => {
        const pillButton = render(<PillButton />);
        expect(pillButton).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<PillButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onPress callback when button is pressed", () => {
        const onPressMock = jest.fn();
        const { getByTestId } = render(<PillButton onPress={onPressMock} />);
        const buttonContainerOuterLayer = getByTestId(
            "button-container-outer-layer"
        );
        const buttonContainer = buttonContainerOuterLayer.findByProps({
            testID: "button-container",
        });
        const button = buttonContainer.findByProps({ testID: "button" });
        fireEvent.press(button);
        expect(onPressMock).toHaveBeenCalled();
    });

    it("renders disabled button correctly", () => {
        const { getByTestId } = render(<PillButton disabled={true} />);
        const buttonContainerOuterLayer = getByTestId(
            "button-container-outer-layer"
        );
        const buttonContainer = buttonContainerOuterLayer.findByProps({
            testID: "button-container",
        });
        const button = buttonContainer.findByProps({ testID: "button" });
        expect(button.props.disabled).toBe(true);
    });

    it("displays label prop", () => {
        const { queryByText } = render(<PillButton label="Meow!" />);
        expect(queryByText("Meow!")).not.toBeNull();
    });
});

describe("<TextButton />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<TextButton />);
        expect(getByText("Button")).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<TextButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onPress callback when button is pressed", () => {
        const onPressMock = jest.fn();
        const { getByText } = render(<TextButton onPress={onPressMock} />);
        const button = getByText("Button");
        fireEvent.press(button);
        expect(onPressMock).toHaveBeenCalled();
    });

    it("renders disabled button correctly", () => {
        const { getByTestId } = render(<TextButton disabled={true} />);
        const buttonContainerOuterLayer = getByTestId(
            "button-container-outer-layer"
        );
        const buttonContainer = buttonContainerOuterLayer.findByProps({
            testID: "button-container",
        });
        const button = buttonContainer.findByProps({ testID: "button" });
        expect(button.props.disabled).toBe(true);
    });

    it("displays label prop", () => {
        const { queryByText } = render(<TextButton label="Click Me!" />);
        expect(queryByText("Click Me!")).not.toBeNull();
    });
});

describe("<FilterButton />", () => {
    it("renders successfully", () => {
        const { getByText } = render(<FilterButton />);
        expect(getByText("First")).toBeDefined();
        expect(getByText("Second")).toBeDefined();
    });

    it("renders correctly", () => {
        const tree = render(<FilterButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onValueChange callback when a button is pressed", () => {
        const onValueChangeMock = jest.fn();
        const { getByText } = render(
            <FilterButton onValueChange={onValueChangeMock} />
        );
        const firstButton = getByText("First");
        fireEvent.press(firstButton);
        expect(onValueChangeMock).toHaveBeenCalledWith("First");

        const secondButton = getByText("Second");
        fireEvent.press(secondButton);
        expect(onValueChangeMock).toHaveBeenCalledWith("Second");
    });

    it("renders disabled button correctly", () => {
        const { getByRole } = render(
            <FilterButton
                firstValue="First"
                secondValue="Second"
                disabled={true}
            />
        );

        const firstButton = getByRole("button", { name: "First" });
        const secondButton = getByRole("button", { name: "Second" });

        expect(firstButton.props.accessibilityState.disabled).toBe(true);
        expect(secondButton.props.accessibilityState.disabled).toBe(true);
    });

    it("displays custom values for buttons", () => {
        const { queryByText } = render(
            <FilterButton firstValue="Option 1" secondValue="Option 2" />
        );
        expect(queryByText("Option 1")).not.toBeNull();
        expect(queryByText("Option 2")).not.toBeNull();
    });
});
