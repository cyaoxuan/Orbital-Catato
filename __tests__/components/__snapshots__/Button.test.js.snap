// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButton /> renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "margin": 8,
        "width": "80%",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#05668D",
          "borderBottomRightRadius": 0,
          "borderColor": "#05668D",
          "borderRadius": 20,
          "borderRightWidth": 0,
          "borderTopRightRadius": 0,
          "borderWidth": 1,
        },
        {
          "borderStyle": "solid",
          "flex": 1,
          "minWidth": 76,
        },
        [
          undefined,
          {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": true,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderBottomRightRadius": 0,
            "borderRadius": 20,
            "borderRightWidth": 0,
            "borderTopRightRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "color": "white",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        >
          First
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderBottomLeftRadius": 0,
          "borderColor": "#05668D",
          "borderRadius": 20,
          "borderTopLeftRadius": 0,
          "borderWidth": 1,
        },
        {
          "borderStyle": "solid",
          "flex": 1,
          "minWidth": 76,
        },
        [
          undefined,
          {},
        ],
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": false,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderBottomLeftRadius": 0,
            "borderRadius": 20,
            "borderTopLeftRadius": 0,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 9,
            },
            {
              "paddingVertical": 9,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.1,
                "lineHeight": 20,
              },
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "color": "#05668D",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
              ],
            ]
          }
        >
          Second
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<PillButton /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "margin": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#05668D",
        "borderRadius": 50,
        "height": 40,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "90%",
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#05668D",
          "borderColor": "transparent",
          "borderRadius": 50,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": 1,
          "justifyContent": "center",
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 50,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 12,
                  },
                  undefined,
                  false,
                  {
                    "color": "white",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                    "fontSize": 16,
                  },
                ],
              ]
            }
            testID="button-text"
          >
            Button
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<TextButton /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": "90%",
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": 1,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  {
                    "marginHorizontal": 12,
                  },
                  undefined,
                  false,
                  {
                    "color": "#FBAB4B",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                    "fontSize": 16,
                  },
                ],
              ]
            }
            testID="button-text"
          >
            Button
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
